src/main/java/com/blackducksoftware/integration/hub/builder/HubCredentialsBuilder.java

```diff
@@ -0,0 +1,131 @@
+import com.blackducksoftware.integration.hub.global.HubCredentialsFieldEnum;
+
+public class HubCredentialsBuilder extends AbstractBuilder<GlobalFieldKey, HubCredentials> {
+
+	private String username;
+	private String password;
+	private int passwordLength;
+
+	public HubCredentialsBuilder() {
+		super(false);
+	}
+
+	public HubCredentialsBuilder(final boolean shouldUseDefaultValues) {
+		super(shouldUseDefaultValues);
+	}
+
+	@Override
+	public ValidationResults<GlobalFieldKey, HubCredentials> build() {
+		final ValidationResults<GlobalFieldKey, HubCredentials> result = assertValid();
+		HubCredentials creds = null;
+		if (StringUtils.isNotBlank(password) && passwordLength == 0) {
+			// Password needs to be encrypted
+			String encryptedPassword = null;
+			try {
+				encryptedPassword = PasswordEncrypter.encrypt(password);
+			} catch (final EncryptionException e) {
+				result.addResult(HubCredentialsFieldEnum.PASSWORD,
+						new ValidationResult(ValidationResultEnum.ERROR, e.getMessage(), e));
+			}
+			creds = new HubCredentials(username, encryptedPassword, password.length());
+		} else {
+			// password is blank or already encrypted so we just pass in the
+			// values given to us
+			creds = new HubCredentials(username, password, passwordLength);
+		}
+
+		result.setConstructedObject(creds);
+		return result;
+	}
+
+	@Override
+	public ValidationResults<GlobalFieldKey, HubCredentials> assertValid() {
+		final ValidationResults<GlobalFieldKey, HubCredentials> result = new ValidationResults<GlobalFieldKey, HubCredentials>();
+
+		validateCredentials(result);
+
+		return result;
+	}
+
+	public void validateCredentials(final ValidationResults<GlobalFieldKey, HubCredentials> result) {
+
+		validateUsername(result);
+		validatePassword(result);
+	}
+
+	public void validateUsername(final ValidationResults<GlobalFieldKey, HubCredentials> result) {
+		if (StringUtils.isBlank(username)) {
+			result.addResult(HubCredentialsFieldEnum.USERNAME,
+					new ValidationResult(ValidationResultEnum.ERROR, "No Hub Username was found."));
+		} else {
+			result.addResult(HubCredentialsFieldEnum.USERNAME, new ValidationResult(ValidationResultEnum.OK, ""));
+		}
+	}
+
+	public void validatePassword(final ValidationResults<GlobalFieldKey, HubCredentials> result) {
+		if (StringUtils.isBlank(password)) {
+			result.addResult(HubCredentialsFieldEnum.PASSWORD,
+					new ValidationResult(ValidationResultEnum.ERROR, "No Hub Password was found."));
+		} else {
+			result.addResult(HubCredentialsFieldEnum.PASSWORD, new ValidationResult(ValidationResultEnum.OK, ""));
+		}
+	}
+
+	public String getUsername() {
+		return username;
+	}
+
+	public void setUsername(final String username) {
+		this.username = username;
+	}
+
+	public String getPassword() {
+		return password;
+	}
+
+	public void setPassword(final String password) {
+		this.password = password;
+	}
+
+	public int getPasswordLength() {
+		return passwordLength;
+	}
+
+	/**
```

Comment:
probably fine for now, but we should consider a setEncryped and setDecrypted method and omit the setPassword and keep this passwordLength field completely internal to this class